libbeat/processors/add_kubernetes_metadata/kubernetes.go

```diff
@@ -171,7 +171,28 @@ func (k *kubernetesAnnotator) init(config kubeAnnotatorConfig, cfg *common.Confi
 			return
 		}
 
-		metaGen := metadata.NewPodMetadataGenerator(cfg, watcher.Store(), nil, nil)
+		metaConf := metadata.GetDefaultResourceMetadataConfig()
+
+		options := kubernetes.WatchOptions{
+			SyncTimeout: config.SyncPeriod,
+			Node:        "",
+		}
+		if config.Namespace != "" {
+			options.Namespace = config.Namespace
+		}
+		nodeWatcher, err := kubernetes.NewWatcher(client, &kubernetes.Node{}, options, nil)
+		if err != nil {
+			k.log.Errorf("couldn't create watcher for %T due to error %+v", &kubernetes.Node{}, err)
+		}
+		namespaceWatcher, err := kubernetes.NewWatcher(client, &kubernetes.Namespace{}, kubernetes.WatchOptions{
+			SyncTimeout: config.SyncPeriod,
+		}, nil)
+		if err != nil {
+			k.log.Errorf("couldn't create watcher for %T due to error %+v", &kubernetes.Namespace{}, err)
+		}
+
+		metaGen := metadata.GetPodMetaGen(cfg, watcher, nodeWatcher, namespaceWatcher, metaConf)
```

Comment:
This code is now quite the same as the one in autodiscover's `NewPodEventer`, maybe we can reuse some things. But to be done in a follow-up, no need to do it here.

Reply:
Added a TODO. Will open a follow-up for this!